src/Commands/GenerateModule.php

```diff
@@ -0,0 +1,325 @@
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/Form/ChoiceProvider", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/Form/Type", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/img/banner", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/img/gerant", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/img/slider", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/Model", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/Presenter", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/src/Repository", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/views/js/form", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/views/js/grid", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/views/public", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/views/templates/admin", 0755);
+            $filesystem->mkdir(_PS_MODULE_DIR_."/".$fileName."/views/templates/hook", 0755);
+            $rows = 24;
+            $progressBar = new ProgressBar($output, $rows);
+            $progressBar->setFormat(
+                "<fg=white;bg=cyan> %status:-45s%</>\n%current%/%max% [%bar%] %percent:3s%%"
+            );
+            $progressBar->setBarCharacter('<fg=magenta>=</>');
+            $progressBar->setProgressCharacter("<fg=green>➤</>");
+            $progressBar->start();
+            for ($i = 0; $i<$rows; $i++) {
+                if ($i < 2) {
+                    $progressBar->setMessage("Starting...", 'status');
+                } elseif ($i < 22) {
+                    $progressBar->setMessage("In progress...", 'status');
+                } else {
+                    $progressBar->setMessage("Almost finished...", 'status');
+                }
+                $progressBar->advance();
+                usleep(20000);
+            }
+            $progressBar->finish();
+            $io->newLine(2);
+            $io->success('Generation finished');
+        } catch (IOExceptionInterface $exception) {
+            $io->error('An error as occurred in the directory path at '.$exception->getPath());
+        }
+
+        $io->text(sprintf("\033\143"));
+        $io->section('Menu for generating hooks');
+        $this->menuHook($io);
+
+        $hookStringInstall="";
+        foreach(self::$tabHook as $hookname){
+            $hookStringInstall .= "
+            && \$this->registerHook('".$hookname."')";
+        }
+        $hookStringFunction="";
+        foreach(self::$tabHook as $hookfunction){
+            $hookfunction = ucfirst($hookfunction);
+            $hookStringFunction .= "
+    public function hook".$hookfunction."(\$params) {
+        //TODO
+    }
+    ";
+        }
+
+        $stringEntryPoint = "";
+        $stringEntryPoint.= "<?php

```

Comment:
The module content is very long, what about placing it in a specific file outside the classs ?

Reply:
I switch to french guys.
Pour le template de génération du module, il va y avoir le nom de la société, l'entête avec copyright, la licence. Chaque société va avoir ses spécificités.
J'ai imaginé proposer un ou deux modèles de base (avec des {moduleName} {modulepath] à remplacer), que chaque agence peut dupliquer et changer.
Par exemple un dossier template avec :
- fop-module.php
- fop-module-widget.php

Lors de la création du module, la command line propose le choix parmi les 2 modèles.

L'agence peut rajouter des modèles dans ce dossier :
- crea-module.php
- crea-module-payment.php, ...

Ces 2 modèles sont proposés lors de la création en command line.

Reply:
Pour moi, la commande de base créé juste la structure. Puis on a des commandes pour rajouter un hook, rajouter un controller, ...

Reply:
Je vois ce que tu veux dire du coup il faut que j'externalise la création du code dans le fichier ?

Reply:
Oui, voila. Ce sera plus souple et plus adaptable.

Reply:
like i said on the slack, we should require the symfony maker-bundle and base our logic on theirs using their generator and everything, that will avoid us to re invent the wheel.

Reply:
> like i said on the slack, we should require the symfony maker-bundle

That's why I've asked to the Core team to add it on the Core because it's really painful - even if we require maker bundle in a module - to get Maker works "as it". For now, we don't have it and this won't happens before 1.7.8 so it will probably never will 👍 .

Reply:
Thanks @mickaelandrieu